### PR TITLE
[docs] Updating README with Quick setup and configuration instructions

### LINKED ISSUE #13: Cleanup tasks
* Remove unused imports
* Document how to configure Firebase, domain, etc.
* Remove hardcoded "desolate.space" domain from Provider example
* Add prettify config and take a pass at entire repo

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,148 @@
-# solana-auth
+# `@crossmint/solana-auth`
 
-Implement passwordless sign-in to your app using Solana wallets (Phantom, sollet, etc.), so you can authenticate and authorize access within your app based on a Solana address. For example, allow customers to access a protected website only if they're whitelisted or if they possess an NFT of yours.
+> Implement passwordless sign-in to your app using Solana wallets (Phantom, sollet, etc.), so you can authenticate and authorize access within your app based on a Solana address. For example, allow customers to access a protected website only if they're whitelisted or if they possess an NFT of yours.
```
Comment (alfonso-paella) [inline README.md]: I think we can get rid of everything from here til line 10 no?

File: README.md [hunk 2]
```diff
@@ -1,10 +1,148 @@
...
+yarn dev
+```
+
+## 🛠 Setup (React + Firebase)
+
+## Install
+
+```sh
+yarn add @crossmint/solana-auth-base@beta  \
+         @crossmint/ solana-auth-react-ui@beta
```
Comment (alfonso-paella) [inline README.md]: I think there's an extra space here

File: README.md [hunk 3]
```diff
@@ -1,10 +1,148 @@
...
+```
+
+Define `signIn` and `signOut`. Here we use firebase utils and `signInWithCustomToken`.
```
Comment (alfonso-paella) [inline README.md]: Proposal:
```
Define what happens when a user is signed in and signed out by implementing `signIn` and `signOut`. 

Here we use those events to handle a session using firebase-auth by using [custom auth](https://firebase.google.com/docs/auth/web/custom-auth).
 ```

File: README.md [hunk 4]
```diff
@@ -1,10 +1,148 @@
...
+    >
+      <Component {...pageProps} />
+    </SolanaAuthProvider>
+  );
+};
+
+export default App;
+```
+
+Use the `SolanaFirebaseAuth` component to render the signin button in your app.
```
Comment (alfonso-paella) [inline README.md]: You say SolanaFirebaseAuth but then use SolanaAuthButtonFirebase

Which one is correct?

File: README.md [hunk 5]
```diff
@@ -1,10 +1,148 @@
...
+
+```js
```
Comment (alfonso-paella) [inline README.md]: I think this should be:

\`\`\`jsx

File: README.md [hunk 6]
```diff
@@ -1,10 +1,148 @@
...
+        <SolanaAuthButtonFirebase auth={auth} />
+      </main>
+    </div>
+  );
+};
+
+export default Index;
+```
+
+## Use the serverless functions.
```
Comment (alfonso-paella) [inline README.md:99]: I think the important thing about this step is that it must match the API routes you used earlier when configuring the provider, right?

File: README.md [hunk 7]
```diff
@@ -1,10 +1,148 @@
...
+import firebaseApp from "../../../utils/firebase";
+
+export default SolanaAuth({ adapter: FirebaseAdapter(firebaseApp) });
+```
+
+This object exposes `getAuthChallenge` and `completeAuthChallenge`. These functions take a request and response like vercel or firebase functions and return `Promise<void>`.
+
+## Client Usage
+
+export default Index;
```
Comment (alfonso-paella) [inline README.md]: Not sure if this line and the next one are intentional?

Comment (alfonso-paella) [reply README.md]: Or maybe we're missing something

File: README.md [hunk 8]
```diff
@@ -1,10 +1,148 @@
...
+
+````
+
+# Custom Configuration
+
+## Defining an adapter
```
Comment (alfonso-paella) [inline README.md]: Defining a database adapter

File: README.md [hunk 9]
```diff
@@ -1,10 +1,148 @@
...
+
```
Comment (alfonso-paella) [inline README.md:117]: Database adapters are needed in order to store transient log-in attempt information. We provide a built-in adapter for Firebase Firestore, or you can implement your own.

File: README.md [hunk 10]
```diff
@@ -1,10 +1,148 @@
...
+**If you are using Firebase, we have built a database adapter for you, accesible with `@crossmint/solana-auth-base`**
+
+Database adapters must fit the `Adapter` shape
```
Comment (alfonso-paella) [inline README.md]: `Adapter` interface

File: README.md [hunk 11]
```diff
@@ -1,10 +1,148 @@
...
+
+```js
```
Comment (alfonso-paella) [inline README.md]: \`\`\`ts

File: README.md [hunk 12]
```diff
@@ -1,10 +1,148 @@
...
+  getNonce(pubkey: string): Promise<any>;
+  generateToken(pubkey: string): Promise<string>;
+  getTLL(pubkey: string): Promise<number>;
+}
+````
+
+A custom database adapter can be passed to the `SolanaAuth` constructor like so:
+
+```js
+SolanaAuth({ adapter: CustomAdapter() });
```
Comment (alfonso-paella) [inline README.md:160]: I would add an example declaration of a custom adapter. For ex:

class CustomAdapter implements Adapter {
  saveSigninAttempt(attempt: SignInAttempt): Promise<void> {
    // TODO: save attempt object into persistent storage. Use the <public key> as primary key 
    // and replace any existing entry with the same key
  }
  getNonce(pubkey: string): Promise<any> {
    // TODO: retrieve the nonce of the most recent sign-in attempt for a public key
  }
  generateToken(pubkey: string): Promise<string> {
    // TODO (Ben) explain this one
  }
  getTLL(pubkey: string): Promise<number> {
      // TODO (Ben) explain this one
  }
}

**Review by alfonso-paella — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jiito) [reply README.md]: Yes. It's redundant right now.

**Comment by ghost** [comment]
@alfonso-paella Updated the docs to include more info on the database adapters and updated language.

**Review by alfonso-paella — APPROVED** [review]

**Pull request merged**